### DISCUSSION TITLE
Help passing fields param in /json endpoint

### DISCUSSION THREAD

**Discussion by cerebrosoft** [brianvoe/gofakeit#276] [Q&A]
I am struggling to get the correct format to create a custom json format from the /json endpoint in gofakeitserver
I've tried many variations but in the end I'm guessing and failing at how I should pack multiple fields in the "fields" GET parameter... e.g. one attempt of many:
http://localhost:8080/json?fields=[{"Name": "first_name", "Function":"firstname"}]

I would be grateful for a small working example!

**Comment by brianvoe** [comment]
I dont know if i fully understand what your trying to do. But if you want to send me a pr with tests that you think it should pass ill make sure that gofakeitserver is able to do pass those tests.

**Reply by cerebrosoft** [reply]
Via the http server, I am trying to use the /json endpoint that generates an object or an array of objects in json format but I cannot figure out what the URL should be to specify the fields I'd like.

the /list endpoint shows a nice example of a response from this endpoint, but I can'f find an example of the request that would generate that sort of response.

From the docs I can determine that the pattern is something like this:

http://localhost:8080/json?fields=<???>,type=object

but I am struggling to figure out what goes in <???>, it appears that it needs to be some complex object, e.g. a list of function and name tuples.

I hope that makes sense!  Thanks.

**Reply by cerebrosoft** [reply]
I finally figured it out- you can pass multiple params, each with the name "fields"

http://localhost:8080/json?fields={"name":"vehicle","function":"car"}&fields={"name":"location","function":"address"}&fields={"name":"description","function":"hipsterparagraph"}

I have not figured out how to incorporate a function like intrange into this sort of request because it doesn't seem to pick up the "min" and "max" values.
